src/util.rs

```diff
@@ -142,15 +156,21 @@ pub(crate) mod ptr {
             Ptr { ptr, _lifetime: PhantomData }
         }
 
+        /// Forget that `Ptr` is known to be validly-aligned for `T`.
+        pub(crate) fn forget_aligned(self) -> Ptr<'a, T, false> {
+            Ptr { ptr: self.ptr, _lifetime: self._lifetime }
```

Comment:
Safety comment? I know this is silly, but something as basic as "the invariants when `ASSUME_ALIGNED = false` are a subset of the invariants when `ASSUME_ALIGNED = true`" should be fine.